webview/src/experiments/components/table/Cell.tsx

```diff
@@ -129,27 +130,38 @@ export const RowActions: React.FC<RowActionsProps> = ({
 
 export const FirstCell: React.FC<
   CellProp &
-    RowActionsProps & {
+    RowActionsProps &
+    WithTableRoot & {
       bulletColor?: string
       toggleExperiment: () => void
+      isRowSelected: boolean
+      toggleRowSelection: () => void
+      toggleStarred: () => void
     }
-> = ({ cell, bulletColor, toggleExperiment, ...rowActionsProps }) => {
+> = ({ cell, bulletColor, toggleExperiment, root, ...rowActionsProps }) => {
   const { row, isPlaceholder } = cell
   const {
     original: { queued }
   } = row
-
   const {
     subRowStates: { plotSelections }
   } = rowActionsProps
 
+  const [ref, needsShadow] = useInView({
+    root,
+    rootMargin: '0px 0px 0px -15px',
+    threshold: 1
+  })
```

Comment:
Are we able to do this without adding an observer to every row's first cell?

Reply:
Good point! This solution requires us to add multiple observers which isn't optimal 🤔  I'll look into a way where we'd only have to rely on one observer!

Reply:
Updated the solution! Now only using one observer and ended up touching less files!